src/utils/utils.ts

```diff
@@ -69,3 +69,83 @@ export const createDatesArray = (
   }
   return datesArray
 }
+
+interface CleanedEarn {
+  _id: number
+  data: [
+    {
+      minFee: number
+      maxFee: number
+      dayCount: number
+      memCount: number
+      minDelay: number
+      maxDelay: number
+      minMinutes: number
+      maxMinutes: number
+    }
+  ]
+}
+
+interface EarnAllInfo {
+  _id: number
+  data: [
+    {
+      minFee: number
+      maxFee: number
+      dayCount: number
+      memCount: number
+      minDelay: number
+      maxDelay: number
+      minMinutes: number
+      maxMinutes: number
+    }
+  ]
+  sortedData: {
+    zeroToOne: number
+    oneToTwo: number
+    twoToThree: number
+    threeToTen: number
+  }
+}
+
+export const earnDataParser = (
+  preParsedData: CleanedEarn
+): EarnAllInfo | Error => {
+  let zeroToOne
+  let oneToTwo
+  let twoToThree
+  let threeToTen
+  preParsedData.data.forEach(feeObject => {
+    if (feeObject.minDelay === 0 && feeObject.maxDelay === 1) {
+      zeroToOne = feeObject.maxFee
+    } else if (feeObject.minDelay === 1 && feeObject.maxDelay === 2) {
+      oneToTwo = feeObject.maxFee
+    } else if (
+      feeObject.minDelay === 2 &&
+      feeObject.maxDelay === 3 &&
+      feeObject.maxMinutes === 70
+    ) {
+      twoToThree = feeObject.maxFee
+    } else if (feeObject.minDelay === 3 && feeObject.maxDelay === 10) {
+      threeToTen = feeObject.maxFee
+    }
+  })
+  if (
+    zeroToOne === undefined ||
+    oneToTwo === undefined ||
+    twoToThree === undefined ||
+    threeToTen === undefined
+  ) {
+    throw new Error('Error: Corrupted Data')
+  }
+
+  const output = {
+    sortedData: {
+      zeroToOne: zeroToOne,
+      oneToTwo: oneToTwo,
+      twoToThree: twoToThree,
+      threeToTen: threeToTen
+    }
```

Comment:
In ES6, there is a new shorthand for object property value. This can be rewritten as
`sortedData: { zeroToOne, oneToTwo, twoToThree, threeToTen }`. (Ref: https://alligator.io/js/object-property-shorthand-es6/)